packages/http-server-javascript/src/scripts/scaffold/bin.mts

```diff
@@ -0,0 +1,698 @@
+  mainTsp: "./main.tsp",
+  projectYaml: "./tspconfig.yaml",
+  packageJson: "./package.json",
+  tsConfigJson: "./tsconfig.json",
+  vsCodeLaunchJson: "./.vscode/launch.json",
+  vsCodeTasksJson: "./.vscode/tasks.json",
+} as const;
+
+function getDefaultTsConfig(standalone: boolean) {
+  return {
+    compilerOptions: {
+      target: "es2020",
+      module: "Node16",
+      moduleResolution: "node16",
+      rootDir: "./",
+      outDir: "./dist/",
+      esModuleInterop: true,
+      forceConsistentCasingInFileNames: true,
+      strict: true,
+      skipLibCheck: true,
+      declaration: true,
+      sourceMap: true,
+    },
+    include: standalone
+      ? ["src/**/*.ts"]
+      : ["src/**/*.ts", "tsp-output/@typespec/http-server-javascript/**/*.ts"],
+  } as const;
+}
+
+const VSCODE_LAUNCH_JSON = {
+  configurations: [
+    {
+      type: "node",
+      request: "launch",
+      name: "Launch Program",
+      program: "${workspaceFolder}/dist/src/index.js",
+      preLaunchTask: "npm: build",
+      internalConsoleOptions: "neverOpen",
+    },
+  ],
+};
+
+const VSCODE_TASKS_JSON = {
+  version: "2.0.0",
+  tasks: [
+    {
+      type: "npm",
+      script: "build",
+      group: "build",
+      problemMatcher: [],
+      label: "npm: build",
+      presentation: {
+        reveal: "silent",
+      },
+    },
+  ],
+};
+
+interface ScaffoldingOptions {
+  /**
+   * If true, the project will be generated in the current directory instead of the output directory.
+   */
+  "no-standalone": boolean;
+  /**
+   * If true, writes will be forced even if the file or setting already exists. Use with caution.
+   */
+  force: boolean;
+}
+
+const DEFAULT_SCAFFOLDING_OPTIONS: ScaffoldingOptions = {
+  "no-standalone": false,
+  force: false,
+};
+
+function parseScaffoldArguments(args: string[]): ScaffoldingOptions {
+  let cursor = 2;
+  const options: Partial<ScaffoldingOptions> = {};
+
+  while (cursor < args.length) {
+    const arg = args[cursor];
+
+    if (arg === "--no-standalone") {
+      options["no-standalone"] = true;
+    } else if (arg === "--force") {
+      options.force = true;
+    } else {
+      console.error(`[hsj] Unrecognized scaffolding argument: '${arg}'`);
+      process.exit(1);
+    }
+
+    cursor++;
+  }
+
+  return { ...DEFAULT_SCAFFOLDING_OPTIONS, ...options };
+}
+
+async function confirmYesNo(message: string): Promise<void> {
+  const rl = readline.createInterface({
+    input: process.stdin,
+    output: process.stdout,
+  });
+
+  try {
+    const response = await rl.question(`${message} [y/N] `);
+
+    if (response.trim().toLowerCase() !== "y") {
+      console.error("[hsj] Operation cancelled.");
+      process.exit(0);
+    }
+  } finally {
+    rl.close();
+  }
+}
+
+export async function scaffold(options: ScaffoldingOptions) {
+  if (options.force) {
+    await confirmYesNo(
+      "[hsj] The `--force` flag is set and will overwrite existing files and settings that may have been modified. Continue?",
```

Comment:
This feels like an odd prompt if someone has specified `--force`  I feel like the behavior should be, silently overwrite if --force is specified, and prompt before overwriting if not, thoughts?

I also feel like, at least to this extent, we should all work the same way:
- Provide command line options that allow overriding config settings
- Read the config to lay down scaffolding where it makes sense relative to emitter-output-dir (as you do here)
- Prompt before overwriting, unless --force is specified
- If we *don't* think `--force` is the right option for night having prompts, we should provide one, like `--quiet`, there should be some way for folks to automate creating a new project with scaffolding

Reply:
My thinking here is just that `--force` is potentially extremely destructive. Someone could lose work using the flag so I really want to make double-sure that's what they want me to do before I go and overwrite controllers they've potentially modified.

`--force --quiet` could be a good option for "do this and don't ask."